test/jest/mocks.ts

```diff
@@ -245,3 +248,53 @@ export const getSelectedInternalAccountFromMockState = (
     state.metamask.internalAccounts.selectedAccount
   ];
 };
+
+export function overrideAccountsFromMockState<
+  MockMetaMaskState extends MockState['metamask'],
+>(
+  state: { metamask: MockMetaMaskState },
+  accounts: InternalAccount[],
+  selectedAccountId?: string,
+): { metamask: MockMetaMaskState } {
+  // First, re-create the accounts mapping and the currently selected account.
+  const [{ id: newFirstAccountId }] = accounts;
+  const newSelectedAccount = selectedAccountId ?? newFirstAccountId ?? '';
+  const newInternalAccounts = accounts.reduce(
+    (
+      acc: MetaMaskReduxState['metamask']['internalAccounts']['accounts'],
+      account,
+    ) => {
+      acc[account.id] = account;
+      return acc;
+    },
+    {},
+  );
+
+  // Re-create the keyring mapping too, since some selectors are using their internal
+  // account list.
+  const newKeyrings: MetaMaskReduxState['metamask']['keyrings'] = [];
+  for (const keyring of state.metamask.keyrings) {
+    const newAccountsForKeyring = [];
+    for (const account of accounts) {
+      if (account.metadata.keyring.type === keyring.type) {
+        newAccountsForKeyring.push(account.address);
+      }
+    }
+    newKeyrings.push({
+      type: keyring.type,
+      accounts: newAccountsForKeyring,
+    });
+  }
+
+  return {
+    ...state,
+    metamask: {
+      ...state.metamask,
+      internalAccounts: {
+        accounts: newInternalAccounts,
+        selectedAccount: newSelectedAccount,
+      },
+      keyrings: newKeyrings,
```

Comment:
We will need to also populate the `balances` and `rates` when we use `getMultichainSelectedAccountCachedBalance` and `getMultichainConversionRate`

Reply:
Sounds good for `balances`. However, I think `currencyRates` can be added explicitly by the caller, otherwise I might override his "test rates". WDYT?